src/wallet/wallet.cpp

```diff
@@ -5547,7 +5547,7 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
-void CWallet::GetAddressForMining(MinerAddress &minerAddress)
+void CWallet::GetAddressForMining(std::optional<MinerAddress> &minerAddress)
```

Comment:
Same comment here; I think this API would be much clearer if it returned the miner address instead of taking a mutable optional argument.